essence-of-live-coding/test/Cell.hs

```diff
@@ -23,10 +25,45 @@ import LiveCoding
 
 import qualified Cell.Util
 import qualified Cell.Monad.Trans
+import Control.Exception (ErrorCall)
+import Test.Framework.Providers.HUnit
+import Control.Exception.Base
+import Test.HUnit
 
 test = testGroup "Cell"
   [ testProperty "steps produces outputs"
     $ \(inputs :: [Int]) -> inputs === fst (runIdentity $ steps (id :: Cell Identity Int Int) inputs)
+  , testProperty "traverse' works as expected for Lists"
+    $ \(inputs :: [[Int]]) -> 
+      fmap (fmap (+10)) inputs === fst (runIdentity $ steps (traverse' (arr (+10) :: Cell Identity Int Int)) inputs)
+  , testProperty "traverse' works as expected for Maybes"
+    $ \(inputs :: [Maybe Int]) -> 
+      fmap (fmap (+10)) inputs === fst (runIdentity $ steps (traverse' (arr (+10) :: Cell Identity Int Int)) inputs)
+  , testCase "traverse' by itself does not force the entire list (ArrM)" 
+    $ testError "traverse' is too strict" id
+    $ head $ head $ fst $ runIdentity $ steps (traverse' (arr id) :: Cell Identity [Int] [Int]) [[1, error "Bang !"]]
+  , testCase "traverse' by itself does not force the entire list (Cell)"
+    $ testError "Traverse' is too strict" id
+    $ head $ head $ fst $ runIdentity $ steps (traverse' cellId :: Cell Identity [Int] [Int]) [[1, error "Bang !"]]
+  , testCase "traverse' of a Cell composed with another Cell does not force the entire list"
+    $ testError "Traverse' is too strict" id
+    $ head $ head $ fst $ runIdentity $ steps (traverse' cellId >>> cellId  :: Cell Identity [Int] [Int]) [[1, error "Bang !"]]
   , Cell.Util.test
   , Cell.Monad.Trans.test
-  ]
+  ]  
+
+testError :: String -> (Bool -> Bool) -> a -> IO ()
+testError message alterBoolValue value = do
```

Comment:
`alterBoolValue` is `id` in all usage cases, it seems. Can you remove it?

Reply:
This function is no longer needed.